File: AICodingWeb/aicoding_web/app/models/customer.py
from datetime import datetime
from app import db
from app.models.base import BaseModel
from sqlalchemy.orm import deferred

class Customer(db.Model, BaseModel):
    """Customer model"""
    
    __tablename__ = 'Customers'
    __table_args__ = {'schema': 'Sales'}
    
    CustomerID = db.Column(db.Integer, primary_key=True)
    CustomerName = db.Column(db.String(100), nullable=False)
    BillToCustomerID = db.Column(db.Integer, db.ForeignKey('Sales.Customers.CustomerID'), nullable=False)
    CustomerCategoryID = db.Column(db.Integer, db.ForeignKey('Sales.CustomerCategories.CustomerCategoryID'), nullable=False)
    BuyingGroupID = db.Column(db.Integer, db.ForeignKey('Sales.BuyingGroups.BuyingGroupID'))
    PrimaryContactPersonID = db.Column(db.Integer, db.ForeignKey('Application.People.PersonID'), nullable=False)
    AlternateContactPersonID = db.Column(db.Integer, db.ForeignKey('Application.People.PersonID'))
    DeliveryMethodID = db.Column(db.Integer, db.ForeignKey('Application.DeliveryMethods.DeliveryMethodID'), nullable=False)
    DeliveryCityID = db.Column(db.Integer, db.ForeignKey('Application.Cities.CityID'), nullable=False)
    PostalCityID = db.Column(db.Integer, db.ForeignKey('Application.Cities.CityID'), nullable=False)
    CreditLimit = db.Column(db.Numeric(18, 2))
    AccountOpenedDate = db.Column(db.Date, nullable=False)
    StandardDiscountPercentage = db.Column(db.Numeric(18, 3), nullable=False)
    IsStatementSent = db.Column(db.Boolean, nullable=False)
    IsOnCreditHold = db.Column(db.Boolean, nullable=False)
    PaymentDays = db.Column(db.Integer, nullable=False)
    PhoneNumber = db.Column(db.String(20), nullable=False)
    FaxNumber = db.Column(db.String(20), nullable=False)
    DeliveryRun = db.Column(db.String(5))
    RunPosition = db.Column(db.String(5))
    WebsiteURL = db.Column(db.String(256), nullable=False)
    DeliveryAddressLine1 = db.Column(db.String(60), nullable=False)
    DeliveryAddressLine2 = db.Column(db.String(60))
    DeliveryPostalCode = db.Column(db.String(10), nullable=False)
    DeliveryLocation = deferred(db.Column(db.String))  # Deferred loading for geography column
    PostalAddressLine1 = db.Column(db.String(60), nullable=False)
    PostalAddressLine2 = db.Column(db.String(60))
    PostalPostalCode = db.Column(db.String(10), nullable=False)
    LastEditedBy = db.Column(db.Integer, db.ForeignKey('Application.People.PersonID'), nullable=False)
    ValidFrom = db.Column(db.DateTime, nullable=False)
    ValidTo = db.Column(db.DateTime, nullable=False)
    
    def __repr__(self):
        return f'<Customer {self.CustomerID}>' 

File: AICodingWeb/Models/Customer.cs
using System;
using System.Collections.Generic;

namespace AICodingWeb.Models
{
    public class Customer
    {
        public int CustomerID { get; set; }
        public string? CustomerName { get; set; }
        public int BillToCustomerID { get; set; }
        public int CustomerCategoryID { get; set; }
        public int? BuyingGroupID { get; set; }
        public int PrimaryContactPersonID { get; set; }
        public int? AlternateContactPersonID { get; set; }
        public int DeliveryMethodID { get; set; }
        public int DeliveryCityID { get; set; }
        public int PostalCityID { get; set; }
        public decimal? CreditLimit { get; set; }
        public DateTime AccountOpenedDate { get; set; }
        public decimal StandardDiscountPercentage { get; set; }
        public bool IsStatementSent { get; set; }
        public bool IsOnCreditHold { get; set; }
        public int PaymentDays { get; set; }
        public string? PhoneNumber { get; set; }
        public string? FaxNumber { get; set; }
        public string? DeliveryRun { get; set; }
        public string? RunPosition { get; set; }
        public string? WebsiteURL { get; set; }
        public string? DeliveryAddressLine1 { get; set; }
        public string? DeliveryAddressLine2 { get; set; }
        public string? DeliveryPostalCode { get; set; }
        public string? PostalAddressLine1 { get; set; }
        public string? PostalAddressLine2 { get; set; }
        public string? PostalPostalCode { get; set; }
        public int LastEditedBy { get; set; }
    }
} 